rero_ils_tools/cli/query/query.py

```diff
@@ -0,0 +1,104 @@
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""RERO ILS circulation category command line interface."""
+
+from __future__ import absolute_import, print_function
+
+import json
+import os
+
+import click
+from flask import current_app
+from flask.cli import with_appcontext
+from invenio_db import db
+from invenio_records_rest.utils import obj_or_import_string
+from rero_ils.modules.utils import (JsonWriter,
+                                    get_record_class_from_schema_or_pid_type,
+                                    read_json_record)
+
+
+@click.command('query')
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-o', '--output', 'output', required=True)
+@click.option('-t', '--record_type', 'record_type', is_flag=False,
+              default='item')
+@click.option('-m', '--model', 'model', required=False)
+@click.option('-f', '--full', 'full', is_flag=True, default=False)
+@click.argument('infile', type=click.File('r'))
+@with_appcontext
+def records_query(infile, full, model, record_type, output, verbose):
+    """Query records.
+
+    :param verbose: verbose
+    :param infile: text file containing the query to select records
+    :param output: JSON output file
+    :param record_type: record type as in RECORDS_REST_ENDPOINTS
+    :param full: extract all fields of record
+    :param model: JSON file to list fields to extract or not
+    """
+    click.secho(f'Extract {record_type} records to: {output}', fg='green')
+    outfile = JsonWriter(output)
+
+    record_class = get_record_class_from_schema_or_pid_type(
+        pid_type=record_type)
+
+    search_class = obj_or_import_string(
+        current_app.config
+        .get('RECORDS_REST_ENDPOINTS')
+        .get(record_type, {}).get('search_class'))
+    
+    if not record_class or not search_class:
+        click.secho(f'Invalid record type: {record_type}', fg='red')
+        exit()
+
+    model_json = {'pid': True}
+    if model:
+        model_filename = open(model)
+        model_json = json.load(model_filename)
+    
+    expert_search = infile.readlines()[0]
+    click.secho(f'Using expert search: {expert_search}', fg='green')
+
+    search = search_class()
+    search = search.query('query_string', query=expert_search)
+    search = search.source('pid')
+    click.secho(f'Number of records to extract: {search.count()}', fg='green')
+
+    pids = [item.pid for item in search.scan()]
+    count = 0
+    for pid in pids:
+        try:
+            record = record_class.get_record_by_pid(pid)
+            count += 1
```

Comment:
Normally I prefer to use `for count, hit in enumerate(search.scan(), 1)`. Then I know exactly witch record had a problem and I can also display the count in the error message.